pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -98,6 +98,33 @@ func Execute(configFile io.Reader) {
 		pruneAccessRecords,
 	)
 
+	signatureRouter := app.NewRoute().PathPrefix("/signature/").Subrouter()
+	// FIXME: We need more fine-grained role (image-admin).
+	signatureAccessRecords := func(*http.Request) []auth.Access {
+		return []auth.Access{
+			{
+				Resource: auth.Resource{
+					Type: "signature",
+				},
+				Action: "get",
+			},
+		}
+	}
+
+	// Advertise features supported by OpenShift
+	if app.Config.HTTP.Headers == nil {
+		app.Config.HTTP.Headers = http.Header{}
+	}
+	app.Config.HTTP.Headers.Set("X-Registry-Supports-Signatures", "1")
+
+	app.RegisterRoute(
+		// GET /admin/signatures/<reference>
```

Comment:
@legionus I don't think a full integration is possible unless we patch upstream code. The passthrough makes sense.